ext/mysql2/client.c

```diff
@@ -82,7 +82,35 @@ struct nogvl_select_db_args {
   MYSQL *mysql;
   char *db;
 };
+static VALUE rb_set_ssl_mode_option(VALUE self, VALUE str) {
+  if( mysql_get_client_version() < 50711 ) {
+    rb_warn( "Your mysql client library does not support setting ssl_mode" );
+    return Qnil;
+  }
 
+  GET_CLIENT(self); 
+  if( NIL_P( str ) ) {
+    rb_raise(cMysql2Error, "ssl_mode= takes DISABLED, PREFERRED, REQUIRED, VERIFY_CA, VERIFY_IDENTITY, you passed nil" );
+  }
+  char *val = StringValueCStr( str );
+  unsigned int setting;
+  if( strncasecmp( val, "DISABLED", 9 ) == 0 ) {
```

Comment:
Fair, something similar tot he existing code in https://github.com/brianmario/mysql2/blob/master/lib/mysql2/client.rb#L90?

Do you want something that would accept strings or ints?  Constant-wise, should we name the same as the constants in C, or should we shorten them?